lib/daru/dataframe.rb

```diff
@@ -1492,39 +1396,19 @@ def report_building(b) # :nodoc: #
 
     def sort! vector_order, opts={}
       raise ArgumentError, 'Required atleast one vector name' if vector_order.empty?
-      opts = {
-        ascending: true,
-        handle_nils: false,
-        by: {}
-      }.merge(opts)
 
-      opts[:ascending] = sort_order_array vector_order, opts[:ascending]
-      opts[:handle_nils] = handle_nils_array vector_order, opts[:handle_nils]
-      blocks = create_logic_blocks vector_order, opts[:by], opts[:ascending]
-
-      block = lambda do |r1, r2|
-        # Build left and right array to compare two rows
-        left = build_array_from_blocks vector_order, opts, blocks, r1, r2
-        right = build_array_from_blocks vector_order, opts, blocks, r2, r1
-
-        # Resolve conflict by Index if all attributes are same
-        left << r1
-        right << r2
-        left <=> right
-      end
+      opts = prepare_sort_opts vector_order, opts
+      block = prepare_sort_block vector_order, opts
 
       idx = (0..@index.size-1).sort(&block)
-
-      old_index = @index.to_a
-      self.index = Daru::Index.new(idx.map { |i| old_index[i] })
+      new_index = @index.reorder(idx)
```

Comment:
`idx = (0..@index.size-1).sort(&block)` can be replaced with `idx = @index.size.times.sort(&block)` I think.